src/main/java/com/github/alinaberlin/security/config/JwtAuthorizationFilter.java

```diff
@@ -0,0 +1,69 @@
+package com.github.alinaberlin.security.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.alinaberlin.security.components.JWTEncoderDecoder;
+import io.jsonwebtoken.Claims;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Filter which check if user is authorized to perform the desired operation.
+ */
+@Component
+public class JwtAuthorizationFilter extends OncePerRequestFilter {
+    private final JWTEncoderDecoder jwtEncoderDecoder;
+    private  ObjectMapper mapper;
+
+    public JwtAuthorizationFilter ( JWTEncoderDecoder jwtEncoderDecoder) {
+        this.jwtEncoderDecoder = jwtEncoderDecoder;
+        this.mapper = new ObjectMapper();
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        Map<String, Object> errorDetails = new HashMap<>();
+
+        try {
+            String accessToken = jwtEncoderDecoder.resolveToken(request);
+            if (accessToken == null ) {
+                filterChain.doFilter(request, response);
+                return;
+            }
+            System.out.println("token : "+accessToken);
+            Claims claims = jwtEncoderDecoder.resolveClaims(request);
+
+            if(claims != null & jwtEncoderDecoder.validateClaims(claims)){
+                String email = claims.getSubject();
+                System.out.println("email : "+email);
+                Authentication authentication =
+                        new UsernamePasswordAuthenticationToken(email,"",new ArrayList<>());
+                SecurityContextHolder.getContext().setAuthentication(authentication);
+            }
+
+        }catch (Exception e){
```

Comment:
We should avoid catching the base Exception class. The type of exception itself is useful information in itself. Take a look at this https://medium.com/javarevisited/mastering-exception-handling-in-java-best-practices-and-examples-f49e3dc955d4